### ISSUE TITLE
Path level $ref is not resolved

### ISSUE THREAD

**Issue by mohsen1** [apigee-127/sway#19]
From https://github.com/swagger-api/swagger-editor/issues/611

A simpler YAML to reproduce:

```yaml
---
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /:
   $ref: 'http://localhost:9000/root.json'
 
```

Inside `root.json`:

```
{
    "get": {
        "responses": {
            "200": {
                "description": "OK"
            }
        }
    }
}
```

After calling `SwaggerApi#create` the `api.definition` value is:

```
{
  "swagger": "2.0",
  "info": {
    "version": "0.0.0",
    "title": "Simple API"
  },
  "paths": {
    "/": {
      "$ref": "http://localhost:9000/root.json"
    }
  }
}
```

**Comment by whitlockjc** [comment]
The ref being in there as-is is because it cannot be resolved.  If you look at the `api.references` information, you'll see that `#/paths/~1` could not be resolved.  This is working as designed unless you can tell me that sway is not resolving something it shouldn't.

**Comment by mohsen1** [comment]
Yes you need to spin up a server to serve the json file. It's not resolving even when the url is reachable

**Comment by whitlockjc** [comment]
The default example on http://editor.swagger.io shows this is working, and it appears to be using sway.  It references `http://petstore.swagger.io/v2/swagger.json#/paths/~1pet` on line 41.

**Comment by whitlockjc** [comment]
I'll look into it.  I also found: https://github.com/whitlockjc/json-refs/issues/30 while working on this.

**Comment by whitlockjc** [comment]
Here is the test I'm using going forward:

```yaml
---
swagger: '2.0'
info:
  version: 0.0.0
  title: Simple API
paths:
  /:
   $ref: 'http://petstore.swagger.io/v2/swagger.json#/paths/~1pet'
```

**Comment by whitlockjc** [comment]
I see this all over the editor.swagger.io console:

`Uncaught (in promise) TypeError: Cannot read property 'petstore_auth' of undefined`

Not sure if that's important or not.

**Comment by mohsen1** [comment]
In development version that error is not being thrown and instead shown to user. It's not finding the petstore security definitions in root swagger. Which makes sense. 

I'm still able to reproduce the original error. I just checked, the XHR to `root.json` is being made and it's successful

**Comment by whitlockjc** [comment]
That error was the cause of the problem.  This is an error in sway where we assume `api.resolved.securityDefinitions` is defined and when it's not, we get an error.  When I fix that, remote references work as you'd expect.  I'll fix this and get a release.

I'm not seeing a case where remote references do not work but once we fix this runtime issue, it should be easy to find since all unresolvable references *(for whatever reason)* are flagged as errors and we can see why.

**Comment by whitlockjc** [comment]
I'm not saying that this issue is fixed.  I'm saying that when I fix sway not to bomb with the error above, remote references appear to work.

**Comment by whitlockjc** [comment]
sway@v0.2.2 is out.  Try this and you should get the proper errors instead of a runtime failure.  Come back to me with the unresolvable error information.  *(Disregard the missing references within the remote document as that is handled in json-refs/issues/30)*

**Comment by mohsen1** [comment]
For resolution I'm using this proxy project so I can use web workers. It's a bit more work to update sway but I'll do it today

https://github.com/mohsen1/sway-worker

**Comment by whitlockjc** [comment]
json-refs/issue/30 has been fixed and sway@master is updated to use it.  Once we know if this issue is still valid and if so, what the real problem is, we can fix it and get a new release out.

**Comment by whitlockjc** [comment]
json-refs/issues/30 has been re-opened but should be unrelated to this.

**Comment by whitlockjc** [comment]
Fixed.  json-refs now has tests using the exact same document in the example above and it is working properly.  I will update sway, again, and you'll need to let me know how this impacts things.

**Comment by whitlockjc** [comment]
sway@v0.2.3 has been released with a patched json-refs to better handle references to remote documents with local references.  *(json-refs has tests for remote references with and without fragments)*

**Comment by whitlockjc** [comment]
With the last build of sway, I am no longer able to reproduce any of the issues encountered today.

**Comment by mohsen1** [comment]
Ok, I'll update you when I had a chance to try the new release.
